lib/services/downloads_service.dart

```diff
@@ -1225,6 +1226,36 @@ class DownloadsService {
     }
   }
 
+  Future<void> addDefaultPlaylistInfoDownload() async {
+
+    String? downloadLocation =
```

Comment:
This location code will return null for anyone who has already created a second download location before upgrading, as long as they don't have a default location set, leading to a broken metadata download.  I believe a better approach would be

    String? downloadLocation =
        FinampSettingsHelper.finampSettings.defaultDownloadLocation;
    if (!FinampSettingsHelper.finampSettings.downloadLocationsMap
        .containsKey(downloadLocation)) {
      downloadLocation = null;
    }
    downloadLocation ??= FinampSettingsHelper.finampSettings.internalSongDir.id;

Reply:
Sounds good!